### PR TITLE
Add support for getting templates from Mandrill.

### PR DESCRIPTION
**Pull request by asheshambasta**
We need to merge this into the nix branch and submit an upstream PR that adds nix support with our additional features. 

---

I've been rather conservative with my cleanups to keep the diff from getting large.
Testing against the Mandrill API is not straightforward and I've only been able to test in the REPL.

### CODE REVIEW

File: src/Network/API/Mandrill/Templates.hs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  ) where
+
+import           Network.API.Mandrill.HTTP
+import           Network.API.Mandrill.Settings
+import           Network.API.Mandrill.Templates.Types
+import           Network.API.Mandrill.Types
+import           Network.HTTP.Client
+
+getTemplates :: MandrillKey -> Maybe Label -> Maybe Manager -> IO (MandrillResponse [TemplatesResponse])
+getTemplates k labelFilt = toMandrillResponse GetTemplates (GetTemplatesRq k labelFilt)
```
Comment (BrechtSerckx) [inline src/Network/API/Mandrill/Templates.hs]:
```suggestion
getTemplates k labelFilt = toMandrillResponse GetTemplates $ GetTemplatesRq k labelFilt
```

File: src/Network/API/Mandrill/Templates/Types.hs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import           Data.String                (IsString)
+import qualified Data.Text                  as T
+import           Lens.Micro.TH              (makeLenses)
+
+import           Network.API.Mandrill.Types
+
+-- | Request to get templates
+data GetTemplatesRq = GetTemplatesRq
+    { _gtrq_key   :: MandrillKey -- ^ API Key
+    -- ^ An optional Template label (for searching)
```
Comment (BrechtSerckx) [inline src/Network/API/Mandrill/Templates/Types.hs:33]:
```suggestion
```

duplicate line here

**Review by BrechtSerckx — CHANGES_REQUESTED** [review]
Some small changes

File: src/Network/API/Mandrill/Templates/Types.hs (hunk 2 above)
Comment (asheshambasta) [reply src/Network/API/Mandrill/Templates/Types.hs:33]: This is a stylish-haskell bug AFAICS.

Comment (BrechtSerckx) [reply src/Network/API/Mandrill/Templates/Types.hs:33]: Hmm, that sucks

**Review by BrechtSerckx — APPROVED** [review]

**Comment by asheshambasta** [comment]
@BrechtSerckx fyi: merged since you had approved it.

**Pull request merged**